oak_containers_launcher/src/server.rs

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod proto {
+    pub mod oak {
+        pub mod containers {
+            tonic::include_proto!("oak.containers");
+        }
+    }
+}
+
+use self::proto::oak::containers::{
+    launcher_server::{Launcher, LauncherServer},
+    GetImageResponse,
+};
+use anyhow::anyhow;
+use futures::Stream;
+use std::pin::Pin;
+use tokio_vsock::VsockListener;
+use tonic::{transport::Server, Request, Response, Status};
+
+type GetImageResponseStream = Pin<Box<dyn Stream<Item = Result<GetImageResponse, Status>> + Send>>;
+
+#[derive(Default)]
+struct LauncherServerImplementation {}
+
+#[tonic::async_trait]
+impl Launcher for LauncherServerImplementation {
+    type GetOakSystemImageStream = GetImageResponseStream;
+    type GetContainerBundleStream = GetImageResponseStream;
+
+    async fn get_oak_system_image(
+        &self,
+        _request: Request<()>,
+    ) -> Result<Response<Self::GetOakSystemImageStream>, tonic::Status> {
+        let response_stream = async_stream::try_stream! {
```

Comment:
Interesting; I guess this yields one chunk that is empty as a response.

Side question: If you leave out the `yield`, would the stream immediately terminate with no data?

Reply:
It'd come out as a type error, the macro expects us to actually yield :s.

The macro isn't from tonic, so edges are a bit rough.